model/config.go

```diff
@@ -2326,7 +2326,8 @@ func (ss *ServiceSettings) isValid() *AppError {
 		}
 	}
 
-	if len(*ss.ListenAddress) == 0 {
+	listenAddressRE := regexp.MustCompile(`^(\d{1,3}.\d{1,3}.\d{1,3}.\d{1,3})?:\d{1,5}$`)
```

Comment:
I would say it is enough, but you can still use 999.999.999.999 as IP which is clearly a not valid ip, or a :99999 which isn't a valid port. You can see more restrictive examples for IP address here: https://www.regular-expressions.info/ip.html